src/mlpack/methods/ann/dists/normal_distribution.hpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file normal_distribution.hpp
+ * @author Atharva Khandait
+ *
+ * Definition of the Normal distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "../activation_functions/softplus_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Multiple univariate Normal(Gaussian) distributions.
+ *
+ * Note: For multivariate distributions, where covariace is not diagonal, there is
+ * another class GaussianDistribution in core/dists.
+ *
+ * @tparam DataType Type of the input data. (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <typename DataType = arma::mat>
+class NormalDistribution
+{
+ public:
+  /**
+   * Default constructor, which creates a Normal distribution with zero
+   * dimension.
+   */
+  NormalDistribution();
+
+  /**
+   * Create multiple Normal distributions with the given parameters.
+   *
+   * @param mean The DataType of means of the multiple distributions
+   * @param stdDev The DataType of standard deviations of the multiple
+   *        distributions.
+   */
+  NormalDistribution(const DataType&& mean, const DataType&& stdDev);
+
+  /*
+   * Create multiple Normal distributions with the given data.
+   * Originally designed to be used along with the ReconstructionLoss class.
+   * The target of the loss function which will be a single matrix can directly
+   * be passed on to this function.
+   *
+   * @param param The DataType which has means in the lower half.
+   *        The upper half has pre standard deviations.
+   * @param applySoftplus If true, after applying softplus function to the pre
+   *        standard deviations, we get the standard deviations.
+   */
+  NormalDistribution(const DataType&& param, const bool applySoftplus = true);
+
+  /**
+   * Return the probabilities of the given DataType of observations.
+   */
+  double Probability(const DataType&& observation) const
+  {
+    return exp(LogProbability(observation));
```

Comment:
It might be usefull to use ``std::exp`` here, to avoid any confusion with ``arma::exp``.

Reply:
Okay, will change this.